keychain/src/keychain.rs

```diff
@@ -42,29 +43,42 @@ impl From<extkey::Error> for Error {
 	}
 }
 
-
 #[derive(Clone, Debug)]
 pub struct Keychain {
 	secp: Secp256k1,
 	extkey: extkey::ExtendedKey,
-
-	/// for tests and burn only, associate the zero fingerprint to a known
-	/// dummy private key
-	pub enable_burn_key: bool,
+	key_overrides: HashMap<Identifier, SecretKey>,
 }
 
 impl Keychain {
 	pub fn root_key_id(&self) -> Identifier {
 		self.extkey.root_key_id.clone()
 	}
 
+	//
+	// For tests and burn only, associate a key identifier with a known secret key.
+	//
+	// TODO - does this need to be a "known" key? Or can we just generate one randomly
+	// and *really* burn the coins?
```

Comment:
I think we could have a use for both. The advantage of a known key in testnet is we can generate a bunch of these outputs with different amounts and easily claim them all at once to test large input set transactions and pruning.